alternator/server.cc

```diff
@@ -408,7 +408,11 @@ future<executor::request_return_type> server::handle_api_request(std::unique_ptr
     auto leave = defer([this] () noexcept { _pending_requests.leave(); });
     //FIXME: Client state can provide more context, e.g. client's endpoint address
     // We use unique_ptr because client_state cannot be moved or copied
-    executor::client_state client_state{executor::client_state::internal_tag()};
+    const bool authenticated = (!username.empty() && username != "<unauthenticated request>");
```

Comment:
This `"<unauthenticated request>"` thing is really funky :-) I wonder if username can be empty and it means unauthenticated, why we needed that special string constant to say the same thing.

If we must have this string constant I think it should be a named constant - I think there's a real risk that someone will change this arbitrary string in server.cc or tracing.hh and not be aware that it needs to be changed in other places.

Reply:
Yes, it was previously only used for tracing and made some sense in there. I'm open to suggestions

Reply:
I personally think it would make more sense to have only username.empty() as the non-authenticated cases, and if we want, when in server.cc we currently do:
```c++
        tracing::set_username(trace_state, auth::authenticated_user(username));
```
we could replace username with something like `username.empty() ? "<unauthenticated request>" : username`.

Reply:
Sounds good, it will limit the scope of this magic constant to just tracing. I'll do that in v2